### PR TITLE
[Bugfix] Fix Disco-CUDAGraph Integration

### PR DESCRIPTION
**Pull request by junrushao**
This PR fixed a bug introduced in #15827 since which the cudagraph's stream is discarded.

### CODE REVIEW

**Comment by junrushao** [comment]
CC: @jinhongyii

**Review by Hzfengsy — APPROVED** [review]

File: src/runtime/disco/nccl/nccl.cc [hunk 1]
```diff
@@ -140,9 +140,19 @@ struct CCLThreadLocalContext {
   deviceStream_t default_stream = nullptr;
   ncclComm_t comm;
 
-  void Clear() { NCCL_CALL(ncclCommDestroy(comm)); }
+  void Clear() {
+    NCCL_CALL(ncclCommDestroy(comm));
+    if (default_stream != nullptr) {
```
Comment (jinhongyii) [inline src/runtime/disco/nccl/nccl.cc:145]: IIUC, default_stream is always nullptr

Comment (tqchen) [reply src/runtime/disco/nccl/nccl.cc:145]: given in this case `default_stream` is a member that can be changed later, i think having such code is better. Note it is not the same as the default stream in the CUDA terminology, but in the nccl runtime

File: src/runtime/disco/nccl/nccl.cc [hunk 2]
```diff
@@ -140,9 +140,19 @@ struct CCLThreadLocalContext {
...
+      StreamDestroy(default_stream);
+    }
+  }
 
-  deviceStream_t GetDefaultStream() { return nullptr; }
+  deviceStream_t GetDefaultStream() {
+    const auto* func = tvm::runtime::Registry::Get("runtime.get_" TVM_DISCO_DEVICE_NAME "_stream");
```
[the remark this answers is not in the document]
Comment (junrushao) [reply src/runtime/disco/nccl/nccl.cc:151]: it's initialized in cudagraph

**Review by jinhongyii — APPROVED** [review]

**Pull request merged**